Add basic scientific names field

jacob-8 opened the pull request:
#### Progress notes

- updated converter test: scn -> scientific_names
- updated entry 
  - be able to edit like a normal text string (saving to database)
  - add convenient italics preview display like we currently have for glosses

- Diego: update translations to include entry.scn field and check table column header
- Diego: update Algolia dashboard to index scn field and test
- Jacob: review and merge PR
  
/// Future
- update to allow for multiple scientific names
- italicize scientific names in table
- add to list view?
- Ask Anna if we can now remove italics authoring from glosses (we won't remove the display of current italics in glosses but this would prohibit proliferation of using scientific names in glosses - is that desired) - ask next group meeting and then remove `field.startsWith('gl')` from `italicizeSelection` button in `EditField.svelte`


#### How can the changes be tested? 
Please also provide applicable links using relative paths from root (e.g. `/apatani/entries`) and reviewers can just add that onto preview urls or localhost.


#### Checklist before marking ready to merge
Please keep it in draft mode until these are completed:
- [ ] Equal time was spent cleaning the code as writing it (Boy Scout Rule)
  - [ ] Functions
    - [ ] Functions that don't belong in Svelte components are extracted out into `.ts` files
    - [ ] Functions are short and well named
    - [ ] Concise tests are written for all functions
  - [ ] Classes (a Svelte Component is a Class)
    - [ ] Svelte components are broken down into smaller components so that each component is responsible for one thing (Single Responsibility Principle)
    - [ ] Stories/variants are written to describe use cases
  - [ ] Comments are only included when absolutely necessary information that cannot be explained in code is needed

Danble reviewed (changes requested):
I already updated locales and added scn to the algolia indexes.
Notes:
Editing a scientific name in the table view saves the values as strings and not as arrays.  Is that ok for the moment? 
That's because we use a different function when saving cells:
![image](https://user-images.githubusercontent.com/43384963/230231446-ba5374ea-f8ae-422c-8461-bd80b1c0a74c.png)
I'm not quite sure if that `saveUpdateToFirestore` should be touched.

jacob-8 commented:
> I already updated locales, added scn to the algolia indexes and fixed table view display of scientific names in italics by default. Notes: Editing a scientific name in the table view saves the values as strings and not as arrays. Is that ok for the moment? That's because we use a different function when saving cells: ![image](https://user-images.githubusercontent.com/43384963/230231446-ba5374ea-f8ae-422c-8461-bd80b1c0a74c.png) I'm not quite sure if that `saveUpdateToFirestore` should be touched.

I updated things to make this work in the last commit
